kbi/0007/index.rst

```diff
@@ -0,0 +1,133 @@
+use a DataLad dataset as means to track the exact state of any downloaded data
+and thereby define a precise starting point for further processing or
+reporting. This is particularly relevant for data files that may change over
+time.
+
+This KBI describes a simple approach how any collection of files that are
+accessible via some URL can be tracked in a DataLad dataset. This approach is
+applicable to a large number of data sources, such as the `Eurostat`_ (the EU's
+portal on statistics and data on Europe) that supports direct file access via
+their `bulk download interface`_.
+
+.. _Eurostat: https://ec.europa.eu/eurostat
+.. _bulk download interface: https://ec.europa.eu/eurostat/data/bulkdownload
+
+
+Solution
+--------
+
+As an example, we will generate a DataLad dataset for a data record in the
+`Gene Expression Omnibus database`_ (GEO), where scientist from around the
+world deposit various kinds of genetics data to facilitate reproducible science
+and future research.
+
+.. _Gene Expression Omnibus database: https://www.ncbi.nlm.nih.gov/geo
+
+We want to create a dataset for the data available under accession number
+GSE209630. GEO offers direct downloads to all four associated files via the
+accession URL https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE209630.
+
+Some of these files are compressed or archives. DataLad provides flexible means
+to transform the original files into a format that is optimally suited for a
+particular type of access or processing. However, here we keep it simple and
+only want to track the files as they are and under the names given by GEO. In
+order to achieve that, we put the filename (that could also specify leading
+sub-directories within the dataset) and the associated access URL as two
+columns in a small `CSV-formatted`_ text file, with one row for each file. The
+resulting table is seen below, first rendered for readability, and afterwards
+the exact same file is shown in its actual CSV text file format.
+
+.. _CSV-formatted: https://en.wikipedia.org/wiki/Comma-separated_values
+
+.. csv-table:: Content of GSE209630_files.csv
+   :file: GSE209630_files.csv
+   :encoding: utf-8
+   :header-rows: 1
+
+.. literalinclude:: GSE209630_files.csv
+
+With this data-specific information ready, we can generate the DataLad dataset
+in two steps automatically.
+
+First we create a new dataset. This step can be skipped, if the files shall be added to an existing dataset.
+
+.. code-block:: console
+
+   % datalad create gse209630
+   create(ok): /tmp/gse209630 (dataset)
+
+We use the ``create`` command and the accession number for the name of the directory that shall contain the newly created DataLad dataset.
+
+Next we can use the ``addurls`` command to populate the dataset with the files
+given in the table. The command requires three parameters: 1) the path to the
+CSV table; 2) the name of the table column with the access URL of a file; and
```

Comment:
```suggestion
CSV table; 2) the name of the table column with the access URL of a file enclosed in curly braces (``{}``); and
```

Reply:
Thanks, I addressed this too with a small edit of your following suggestion.